### PR TITLE
fix(scaletest): deploy external provisionerd

### PR DESCRIPTION
**Pull request by johnstcn**
- Adds an external provisionerd deployment (the lazy way, Helm support is in progress)
- Plumbs through resource values for provisionerd deployment

Other minor fixes:
- Explicitly set secret type so it doesn't get constantly re-created
- Unspecify stable channel so we don't get unexpectedly upgraded

### CODE REVIEW

File: scaletest/terraform/coder.tf [hunk 1]
```diff
@@ -242,6 +245,168 @@ resource "local_file" "kubernetes_template" {
   EOF
 }
 
+# TODO(cian): Remove this when we have support in the Helm chart.
+# Ref: https://github.com/coder/coder/issues/8243
+resource "local_file" "provisionerd_deployment" {
+    filename = "${path.module}/../.coderv2/provisionerd-deployment.yaml"
+    content = <<EOF
```
Comment (mtojek) [inline scaletest/terraform/coder.tf:252]: curious: can this be `source`d from a regular YAML file?

File: scaletest/terraform/gcp_cluster.tf [hunk 2]
```diff
@@ -18,7 +18,7 @@ resource "google_container_cluster" "primary" {
 
   }
   release_channel {
-    channel = "STABLE"
+    channel = "UNSPECIFIED"
```
Comment (mtojek) [inline scaletest/terraform/gcp_cluster.tf:21]: nit: maybe leave a comment here? somebody may not figure it out in the future that this is on purpose.

**Review by mtojek — APPROVED** [review]

File: scaletest/terraform/coder.tf (hunk 1 above)
Comment (johnstcn) [reply scaletest/terraform/coder.tf:252]: I think you can include it as a template, but I find it more straightforward to just inline.

File: scaletest/terraform/gcp_cluster.tf (hunk 2 above)
Comment (johnstcn) [inline scaletest/terraform/gcp_cluster.tf:21]:
```suggestion
    # Setting release channel as STABLE can cause unexpected cluster upgrades.
    channel = "UNSPECIFIED"
```

File: scaletest/terraform/coder.tf (hunk 1 above)
Comment (mtojek) [reply scaletest/terraform/coder.tf:252]: 👍 

My only concern is IDE support, but that is not affecting the product.

File: scaletest/terraform/coder.tf [hunk 3]
```diff
@@ -242,6 +245,168 @@ resource "local_file" "kubernetes_template" {
   EOF
 }
 
+# TODO(cian): Remove this when we have support in the Helm chart.
+# Ref: https://github.com/coder/coder/issues/8243
+resource "local_file" "provisionerd_deployment" {
+  filename = "${path.module}/../.coderv2/provisionerd-deployment.yaml"
+  content  = <<EOF
+apiVersion: apps/v1
```
Comment (mafredri) [inline scaletest/terraform/coder.tf:253]: Consider formatting with prettier, just so it matches our other YAML?

**Review by mafredri — APPROVED** [review]

**Pull request merged**